tests/network/mod.rs

```diff
@@ -59,8 +59,13 @@ impl<D: DistAlgorithm> TestNode<D> {
         &self.algo
     }
 
+    /// Returns the internal algorithm's mutable instance.
+    #[allow(unused)] // Not used in all tests.
+    pub fn instance_mut(&mut self) -> &mut D {
+        &mut self.algo
+    }
     /// Creates a new test node with the given broadcast instance.
-    fn new((algo, step): (D, DaStep<D>)) -> TestNode<D> {
+    pub fn new((algo, step): (D, DaStep<D>)) -> TestNode<D> {
```

Comment:
Do `new` and `dispatch_messages` really need to be public now? They don't seem to be used when restarting node 0.

Reply:
OK, unmaking `new` as `pub`.
`dispatch_messages` is used to dispatch the messages in the step after restarting node 0.